### DISCUSSION TITLE
Shelly 1 with contact sensor on gpio3

### DISCUSSION THREAD

**Discussion by spiccinocchi** [RavenSystem/esp-homekit-devices#2195] [MEPLHAA Scripts]
I need help with the script to configure a Shelly 1 to act as a switch controlled by a button on gpio5 and with a contact sensor accessory with state controlled by gpio3.
Here is me script:

{"c":{"io":[[[4],2],[[3],6,0,1]]},
"a":[{
"0":{"r":[[4]]},
"1":{"r":[[4,1]]},
"b":[[5,0]],
"es": [
        { 
          "t": 5,
          "g1": [[ 3 ]]
        }]
}]}

The switch work with gpio5 but the gpio3 not work and the contact sensor state never change. Can someone help me?

**Comment by RavenSystem** [comment]
{"c":{"io":[[[4],2],[[5],6,0,1],[[3],6,1]]},
"a":[{
"0":{"r":[[4]]},
"1":{"r":[[4,1]]},
"b":[[5,0]],
"es": [
{
"t": 5,
"f0": [[ 3,0 ]],
"f1": [[ 3 ]]
}]
}]}

**Reply by spiccinocchi** [reply]
Thank you for the script. 
I tried it but, using the shelly adapter and connecting together the yellow and black wire nothing happens to the contact sensor state.
Am I doing something wrong?

**Reply by spiccinocchi** [reply]
I find the problem, it was the shelly 1 adapter not working. I changed it and the script work perfectily.
Thank you for the support.
